asciidtype/tests/test_asciidtype.py

```diff
@@ -0,0 +1,20 @@
+import numpy as np
+
+from asciidtype import ASCIIDType, ASCIIScalar
+
+
+def test_dtype_creation():
+    dtype = ASCIIDType(4)
+    assert str(dtype) == "ASCIIDType(4)"
+
+
+def test_scalar_creation():
+    dtype = ASCIIDType(7)
+    ASCIIScalar('string', dtype)
+
+
+def test_creation_with_explicit_dtype():
+    dtype = ASCIIDType(7)
+    arr = np.array(["hello", "this", "is", "an", "array"], dtype=dtype)
+    assert repr(arr) == (
+        "array(['hello', 'this', 'is', 'an', 'array'], dtype=ASCIIDType(7))")
```

Comment:
I was going to say yes, but currently the `dtype=ASCIIDType` isn't passed through correctly, it probably works for `np.core.multiarray._multiarray_umath._discover_array_parameters()`, though.

Passing through the DType type/class rather than just the dtype is something we/I need to look into...  (A similar alternative might be to introduce abstract instances, that cannot be attached to an array, but represent the same thing as ASCIIDType).

Reply:
I opened https://github.com/numpy/numpy/issues/22756 to track this, seems like it would be useful.